docs/DeveloperGuide.md

```diff
@@ -4,30 +4,274 @@
+
+1. **Remove an Event:** When given the name of an event, it searches for and deletes it from the event list if it exists.
+2. **Remove a Participant:** When provided with a participant’s name and the name of an event, it attempts to remove the specified participant from that event.
+
+#### Feature Implementation
+
+Given below is an example usage scenario for the `remove` mechanism, and how it behaves at each step.
+
+1. The user enters the command `remove` followed by `-e` or `-p` to specify removing an event or participant.
+2. This step is determined by our `Parser` which parses through the user input to determine if it is adding a participant or event
+3. Based on the parsed input, `RemoveComamnd` executes one of the following actions:
+   + **Remove Event:** Remove the specified event from `EventList` using the provided event name
+   + **Remove Participant:** Locates the event in `EventList` and deletes the specified participant
+4. If the event or participant is not found, `RemoveCommand` sets a failure message.
+
+The interactions between components during the execution of the `remove` command are show in the **Sequence Diagram** below:
+
+**Remove Event**
+
+<img src = "images/RemoveEventSequenceDiagram.png">
+
+**Remove Participant**
+
+<img src = "images/RemoveParticipantSequenceDiagram.png">
+
+### View feature
+
+The `view` feature allows users to view the list of participants of an event.
+The view command will search for an event by its name and display all its participants if found.
+It is implemented in the `ViewCommand` class which extends the base `Command` class and interacts with the `EventList` to retrieve and display participant information.
+
+The `ViewCommand` class handles one primary function:
+
+1. It is to enable users to view all participants associated with a specific event. 
+It accomplishes this by searching for the event by its name and displaying participant details if the event is found.
+
+#### Feature Implementation
+
+Given below is an example usage scenario for the `view` mechanism, and how it behaves at each step
+
+1. The user enters the command `view` followed by an `-e` flag to indicate which event to view participants from
+2. The command is processed by the `Parser`, which identifies that the user intends to view participants for the specified event.
+3. The `ViewCommand` attempts to retrieve the event from the `EventList` using the `getEventByName` method.
+4. If the event is found:
+   - The `ViewCommand` constructs a success message that includes the number of participants and their details in a formatted manner.
+
+   If the event is not found:
+   - The `ViewCommand` sets an error message.
+5. The message generated by `ViewCommand` is returned to the user interface, which then displays it to the user
+
+The interactions between components during the execution of the `view` command are show in the **Sequence Diagram** below:
+
+<img src = "images/ViewEventSequenceDiagram.png">
```

Comment:
Similar to previous comments regarding `Participant`, if you have another relevant class like `ParticipantList`, it might be worth noting down its details in the DG?

Reply:
Regarding `Participant`, see the above comment on the `Event` component.